plugins/woo-ai/src/product-description/product-description-button-container.tsx

```diff
@@ -1,61 +1,68 @@
 /**
  * External dependencies
  */
-import apiFetch from '@wordpress/api-fetch';
 import { __, sprintf } from '@wordpress/i18n';
 import { useState, useEffect, useRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-import { useTinyEditor } from '../hooks/useTinyEditor';
+import { useTinyEditor, useCompletion } from '../hooks';
 import MagicIcon from '../../assets/images/icons/magic.svg';
 
-type WooApiMessage = {
-	role: string;
-	content: string;
-};
-
-type WooApiResponse = {
-	generated_text: string;
-	previous_messages: WooApiMessage[];
-};
+const DESCRIPTION_MAX_LENGTH = 300;
+const MIN_TITLE_LENGTH = 20;
 
-type WPAPIError = {
-	code: string;
-	message: string;
-	data: {
-		status: number;
+const getApiError = ( status?: number ) => {
+	const errorMessagesByStatus: Record< number, string > = {
+		429: __(
+			'There have been too many requests. Please wait for a few minutes and try again.',
+			'woocommerce'
+		),
+		408: __(
+			'It seems the server is taking too long to respond. This is an experimental feature, so please try again later.',
+			'woocommerce'
+		),
 	};
-};
 
-const generatingContentPhrases = [
-	__( 'Please wait…', 'woocommerce' ),
-	__( 'Just a little longer…', 'woocommerce' ),
-	__( 'Almost there…', 'woocommerce' ),
-];
+	if ( status && errorMessagesByStatus[ status ] ) {
+		return errorMessagesByStatus[ status ];
+	}
 
-const getGeneratingContentPhrase = () =>
-	generatingContentPhrases[
-		Math.floor( Math.random() * generatingContentPhrases.length )
-	];
-
-const DESCRIPTION_MAX_LENGTH = 300;
-const MIN_TITLE_LENGTH = 20;
+	return __(
+		`Apologies, this is an experimental feature and there was an error with this service. Please try again.`,
+		'woocommerce'
+	);
+};
 
 export function WriteItForMeButtonContainer() {
-	const [ fetching, setFetching ] = useState( false );
-	const generatingContentPhraseInterval = useRef< number >();
 	const titleEl = useRef< HTMLInputElement >(
 		document.querySelector( '#title' )
 	);
+	const [ fetching, setFetching ] = useState< boolean >( false );
 	const [ productTitle, setProductTitle ] = useState< string >(
 		titleEl.current?.value || ''
 	);
 	const tinyEditor = useTinyEditor();
+	const { requestCompletion, completionActive, stopCompletion } =
+		useCompletion( {
+			onStreamMessage: ( content ) => {
+				// This prevents printing out incomplete HTML tags.
+				const ignoreRegex = new RegExp( /<\/?\w*[^>]*$/g );
+				if ( ! ignoreRegex.test( content ) ) {
+					tinyEditor.setContent( content );
+				}
+			},
+			onStreamError: ( error ) => {
+				// eslint-disable-next-line no-console
+				console.debug( 'Streaming error encoutered', error );
 
-	const writeItForMeDisabled =
-		fetching || ! productTitle || productTitle.length < MIN_TITLE_LENGTH;
+				tinyEditor.setContent( getApiError() );
```

Comment:
This error handling isn't great at the moment, but I'm uncertain of what the endpoint actually supports (and it sounds like not much so far).

Reply:
Yeah, I'm sure we'll end up changing this anyways to use Gutenberg notices/snackbars once we start to improve the UX.